packages/ui/src/components/Input/textArea.tsx

```diff
@@ -0,0 +1,137 @@
+        const newValue = e.target.value
+
+        if (propsValue === undefined) {
+            setInternalValue(newValue)
+        }
+
+        if (propsOnChange) {
+            propsOnChange(e)
+        }
+    }
+
+    const currentValue = propsValue !== undefined ? propsValue : internalValue
+    const characterCount = `${currentValue.length}/${maxLength}`
+
+    const hasBottomPanelUtils = bottomPanelUtils.length > 0
+    const shouldShowRightPanel = showCount || onSubmit
+    const showBottomPanel = hasBottomPanelUtils || shouldShowRightPanel
+
+    const textareaClassName = classNames(
+        `${TEXTAREA_CLASS_PREFIX}-textarea`,
+        {
+            [`${TEXTAREA_CLASS_PREFIX}-disabled`]: disabled,
+            [`${TEXTAREA_CLASS_PREFIX}-show-bottom-panel`]: showBottomPanel,
+            [`${TEXTAREA_CLASS_PREFIX}-focused`]: autoFocus,
+        },
+        className,
+    )
+
+    return (
+        <div className={`${TEXTAREA_CLASS_PREFIX} ${TEXTAREA_CLASS_PREFIX}-textarea-wrapper`}>
+            <DefaultTextArea
+                {...restProps}
+                ref={ref}
+                prefixCls={TEXTAREA_CLASS_PREFIX}
+                className={textareaClassName}
+                disabled={disabled}
+                autoSize={autoSize}
+                maxLength={maxLength}
+                showCount={false}
+                value={currentValue}
+                onChange={handleChange}
+                autoFocus={autoFocus}
+            />
+
+            {showBottomPanel && (
+                <span className={`${TEXTAREA_CLASS_PREFIX}-bottom-panel`}>
+                    {hasBottomPanelUtils && (
+                        <span className={`${TEXTAREA_CLASS_PREFIX}-utils`}>
+                            {bottomPanelUtils.map((util, index) => (
+                                <React.Fragment key={index}>
+                                    {React.cloneElement(util, { disabled: util.props.disabled || disabled })}
+                                </React.Fragment>
+                            ))}
+                        </span>
+                    )}
+
+                    {shouldShowRightPanel && (
+                        <span className={`${TEXTAREA_CLASS_PREFIX}-bottom-panel-right`}>
+                            {showCount && (
+                                <span className={`${TEXTAREA_CLASS_PREFIX}-count`}>
+                                    {characterCount}
+                                </span>
+                            )}
+
+                            {
+                                onSubmit &&
+                                <Button
+                                    disabled={disabled || isSubmitDisabled}
+                                    type='accent'
+                                    size='medium'
+                                    onClick={() => onSubmit(currentValue)}
+                                    icon={<ArrowRight className='arrow-icon' size='small' />}
```

Comment:
maybe call className rotated-arrow-icon so that it is just a litte bit more obvious

Reply:
That's bad

Reply:
Consider adding new Icon if needed...

Right now it seems like implemented f*ck up of a design team :)